compiler/src/dotty/tools/dotc/sbt/ExtractDependencies.scala

```diff
@@ -274,24 +270,32 @@ private class ExtractDependenciesCollector extends tpd.TreeTraverser { thisTreeT
 
   private def addUsedName(name: Name, scope: UseScope)(implicit ctx: Context): Unit = {
     val fromClass = resolveDependencySource
-    if (fromClass ne NoSymbol) {
+    if (fromClass.exists) {
       assert(fromClass.isClass)
       addUsedName(fromClass, name, scope)
     }
   }
 
+  /** Mangle a JVM symbol name in a format better suited for internal uses by sbt. */
+  private def mangledName(sym: Symbol)(implicit ctx: Context): Name = {
+    def constructorName = sym.enclosingClass.fullName ++ ";init;"
```

Comment:
@smarter I didn't use the exact same scheme as in sbt/zinc#288. Let me know if you think it is an issue

Reply:
I remember weird things happening in zinc when I used "." in the naming scheme, that's why I switched to ";". It would be prudent to do the same here.

Reply:
(I don't remember what the weird things where sadly)